esphome/components/xpt2046/touchscreen/xpt2046.cpp

```diff
@@ -0,0 +1,114 @@
+#include "xpt2046.h"
+#include "esphome/core/log.h"
+#include "esphome/core/helpers.h"
+
+#include <algorithm>
+
+namespace esphome {
+namespace xpt2046 {
+
+static const char *const TAG = "xpt2046";
+
+void XPT2046Component::setup() {
+  if (this->irq_pin_ != nullptr) {
+    // The pin reports a touch with a falling edge. Unfortunately the pin goes also changes state
+    // while the channels are read and wiring it as an interrupt is not straightforward and would
+    // need careful masking. A GPIO poll is cheap so we'll just use that.
+
+    this->irq_pin_->setup();  // INPUT
+    this->irq_pin_->pin_mode(gpio::FLAG_INPUT | gpio::FLAG_PULLUP);
+    this->irq_pin_->setup();
+    this->attach_interrupt_(this->irq_pin_, gpio::INTERRUPT_FALLING_EDGE);
+  }
+  spi_setup();
+  read_adc_(0xD0);  // ADC powerdown, enable PENIRQ pin
```

Comment:
I know you probably just moved this file for the most part, but it shows up as an entirely new file on github due to the changes, so it may as well be cleaned up to standards =)

```suggestion
  this->spi_setup();
  this->read_adc_(0xD0);  // ADC powerdown, enable PENIRQ pin
```